python_modules/dagster/dagster/core/execution/plan/execute_plan.py

```diff
@@ -59,12 +61,27 @@ def inner_plan_execution_iterator(
             )
 
             # capture all of the logs for this step
-            with pipeline_context.instance.compute_log_manager.watch(
-                step_context.pipeline_run, step_context.step.key
-            ):
-                yield DagsterEvent.capture_logs(
-                    step_context, log_key=step_context.step.key, steps=[step_context.step]
-                )
+            with ExitStack() as stack:
+                log_capture_error = None
+                try:
+                    stack.enter_context(
+                        pipeline_context.instance.compute_log_manager.watch(
+                            step_context.pipeline_run, step_context.step.key
+                        )
+                    )
+                except Exception as e:
+                    log_capture_error = e
+                    logging.exception(
+                        "Exception while setting up compute log capture for step %s in run %s: %s",
+                        step_context.step.key,
+                        step_context.pipeline_run.run_id,
+                        e,
+                    )
+
+                if not log_capture_error:
+                    yield DagsterEvent.capture_logs(
+                        step_context, log_key=step_context.step.key, steps=[step_context.step]
+                    )
```

Comment:
just throwing out another option here: we're not getting a ton of value out of watch() being a contextmanager after this change, and there's only one callsite (outside of tests) - should we have a start_watch and end_watch method instead and make it no longer a contextmanager? Seems like that would simplify the logic here a bit

The 'Exception while cleaning up compute log capture' would need to go in a try: finally: block probably - but then we could count on there being one place where the cleanup happens instead of two (one in the try block you added, one where the ExitStack cleans up)

Reply:
I'm fine (actually really psyched to make changes here), but not sure if `ComputeLogManager` is the sort of thing that users have implemented their own implementation of. 

I've been super conservative in the past about making any changes to the API of these configurable classes.

Reply:
watch() isn't even an abstract method so I think it's not particularly likely?